mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +126,31 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    kwargs = {'decode_times': False,
+              'concat_dim': 'Time'}
+
+    softLimit = resource.getrlimit(resource.RLIMIT_NOFILE)[0]
+
+    # use autoclose if we will use more than half the soft limit on open files
+    autoclose = len(fileNames) > softLimit/2
```

Comment:
This makes sense to me.  We may have issues that arise that would require us to use a different constant.  To be safe we could make `2` be a config parameter, something like `autocloseSF = 2` in the `config.default` file.  Then, if there were an issue with too many open files we can give the user some recourse.